tutorials/tts/2_Inference_DurationPitchControl.ipynb

```diff
@@ -0,0 +1,490 @@
+    "\n",
+    "enc_out = enc_out + pitch_emb.transpose(1, 2)  # We add the pitch to the encoder output\n",
+    "spec, *_ = self.decoder(input=len_regulated, seq_lens=dec_lens)  # We send the sum to the decoder to get the spectrogram\n",
+    "```\n",
+    "\n",
+    "Let's see the `pitch_predicted` for a sample text. You can run the below cell. You should get an image that looks like for the input `Hey, what is my pitch?`:\n",
+    "<img src=\"files/fastpitch-pitch.png\">\n",
+    "Notice that the last word `pitch` has an increase in pitch to stress that it is a question."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import librosa\n",
+    "import librosa.display\n",
+    "from matplotlib import pyplot as plt\n",
+    "import numpy as np\n",
+    "from nemo.collections.tts.helpers.helpers import regulate_len\n",
+    "%matplotlib inline\n",
+    "\n",
+    "#Define what we want the model to say\n",
+    "input_string = \"Hey, what is my pitch?\"  # Feel free to change it and experiment\n",
+    "\n",
+    "# Run inference to get spectrogram and pitch\n",
+    "with torch.no_grad():\n",
+    "    spec, audio, durs_pred, pitch_pred = str_to_audio(input_string)\n",
+    "\n",
+    "    # FastPitch predicts one pitch value per token. To plot it, we have to expand the token length to the spectrogram length\n",
+    "    pitch_pred, _ = regulate_len(durs_pred, pitch_pred.unsqueeze(-1))\n",
+    "    pitch_pred = pitch_pred.squeeze(-1)\n",
+    "    # Note we have to unnormalize the pitch with LJSpeech's mean and std\n",
+    "    pitch_pred = pitch_pred * 65.72037058703644 + 214.72202032404294\n",
+    "\n",
+    "# Let's plot the predicted pitch and how it affects the predicted audio\n",
+    "fig, ax = plt.subplots(figsize=(12, 6))\n",
+    "spec = np.abs(librosa.stft(audio[0], n_fft=1024))\n",
+    "ax.plot(pitch_pred.cpu().numpy()[0], color='cyan', linewidth=1)\n",
+    "librosa.display.specshow(np.log(spec+1e-12), y_axis='log')\n",
+    "ipd.display(ipd.Audio(audio, rate=sr))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Plot Control\n",
+    "\n",
+    "Now that we see how the pitch affects the predicted spectrogram, we can now adjust it to add some effects. We will explore 4 different manipulations:\n",
+    "\n",
+    "1) Pitch shift\n",
+    "\n",
+    "2) Pitch flatten\n",
+    "\n",
+    "3) Pitch inversion\n",
+    "\n",
+    "4) Pitch amplification"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First we look at pitch shift. To shift the pitch up or down by some Hz, we can just add or subtract as needed\n",
+    "# Let's shift the pitch down by 50 Hz\n",
+    "# First, let's run the previous example and then shift down\n",
```

Comment:
Can you pull these numbered comments out of the code cells and into text cells? Might be stylistically more consistent for a notebook

Reply:
done